snuba/query/expressions.py

```diff
@@ -417,6 +485,50 @@ def functional_eq(self, other: Expression) -> bool:
                 return False
         return True
 
+    def __gt__(self, other: Any) -> bool:
```

Comment:
This is a lot of code that is duplicated twice. I think you can move all of this into one function, use the `operator` module to collapse this down.

An overly simplified example of what I mean:
```
def _gt_lt_helper(self, other, op, other_op):
    ...
    if op(self.function_name, other.function_name):
        return False
    elif other_op(self.function_name, other.function_name):
        return True
    ...

def __gt__(self, other):
    return self._gt_lt_helper(other, operator.gt, operator.lt)
...
```